### DISCUSSION TITLE
How I find the qemu plugin?

### DISCUSSION THREAD

**Discussion by marmotta77** [BlitterStudio/amiberry#1534] [Q&A]
In the new version is possible to use the ppc emulation but only with qemu-plugin. I have tried the fs-uae version but with no success... I need the ARM64 version.

**Comment by giantclambake** [comment]
Not sure if the library works or not, but you can try -- https://fs-uae.net/download#plugins

**Reply by marmotta77** [reply]
> Not sure if the library works or not, but you can try -- https://fs-uae.net/download#plugins

Already tried. Amiberry search the plugin on app container > framework > plugin > qemu-uae.dylib, but this file is not available anywhere. Fs-uae have a qemu-uae.so, I have put it on: documents > amiberry > plugins with no success :-(

**Reply by giantclambake** [reply]
> I need the ARM64 version

Ahh.... when I see 'ARM64' I don't think macOS but instead RPi   ;)

**Reply by midwan** [reply]
Doesn't it work if you rename the file, and place it in the expected directory?
There is an ARM64 macOS version available in the link posted above, just the filename is different (`.so` is the Linux standard for libraries, in macOS they should be named `.dylib`)

**Reply by marmotta77** [reply]
I have tried, the log show this:

`PPC: Loading QEmu implementation
DLOPEN: dlopen(/Applications/Emulatori/Amiberry.app/Contents/Frameworks/qemu-uae.dylib, 0x0002): tried: '/Applications/Emulatori/Amiberry.app/Contents/Frameworks/qemu-uae.dylib' (code signature invalid in <68F9F8D0-696D-315C-816B-AD56342EC8D9> '/Applications/Emulatori/Amiberry.app/Contents/Frameworks/qemu-uae.dylib' (errno=1) sliceOffset=0x00000000, codeBlobOffset=0x007E81A0, codeBlobSize=0x0001E420), '/System/Volumes/Preboot/Cryptexes/OS/Applications/Emulatori/Amiberry.app/Contents/Frameworks/qemu-uae.dylib' (no such file), '/Applications/Emulatori/Amiberry.app/Contents/Frameworks/qemu-uae.dylib' (code signature invalid in <68F9F8D0-696D-315C-816B-AD56342EC8D9> '/Applications/Emulatori/Amiberry.app/Contents/Frameworks/qemu-uae.dylib' (errno=1) sliceOffset=0x00000000, codeBlobOffset=0x007E81A0, codeBlobSize=0x0001E420)
DLOPEN: Failed to open /Applications/Emulatori/Amiberry.app/Contents/Frameworks/qemu-uae.dylib
PPC: Loading dummy implementation`

Signature invalid?

**Reply by midwan** [reply]
Ah, of course. MacOS security gets in the way here.
We would have to digitally sign the library with the same certificate that Amiberry is signed, otherwise the OS blocks it as potential malware.

I'll take a look at this in RC2 then.

**Reply by marmotta77** [reply]
However, even in FS-UAE the Arm64 plugin does not work (green screen), it is likely to be updated

**Reply by giantclambake** [reply]
@midwan ~ query: doesn't PPC emulation infer JIT to be available as well?

**Reply by midwan** [reply]
No, JIT is not related to PPC in any way.

**Comment by marmotta77** [comment]
On the new RC2

`DLOPEN: dlopen(/Applications/Emulatori/Amiberry.app/Contents/Resources/plugins/qemu-uae.dylib, 0x0002): tried: '/Applications/Emulatori/Amiberry.app/Contents/Resources/plugins/qemu-uae.dylib' (code signature in <68F9F8D0-696D-315C-816B-AD56342EC8D9> '/Applications/Emulatori/Amiberry.app/Contents/Resources/plugins/qemu-uae.dylib' not valid for use in process: mapping process and mapped file (non-platform) have different Team IDs), '/System/Volumes/Preboot/Cryptexes/OS/Applications/Emulatori/Amiberry.app/Contents/Resources/plugins/qemu-uae.dylib' (no such file), '/Applications/Emulatori/Amiberry.app/Contents/Resources/plugins/qemu-uae.dylib' (code signature in <68F9F8D0-696D-315C-816B-AD56342EC8D9> '/Applications/Emulatori/Amiberry.app/Contents/Resources/plugins/qemu-uae.dylib' not valid for use in process: mapping process and mapped file (non-platform) have different Team IDs)
DLOPEN: Failed to open /Applications/Emulatori/Amiberry.app/Contents/Resources/plugins/qemu-uae.dylib
PPC: Loading dummy implementation`

**Reply by midwan** [reply]
Yes, known issue for now. The qemu library needs to be signed with the same certificate as the Amiberry bundle, otherwise MacOS will not allow it to load.

**Reply by marmotta77** [reply]
> Yes, known issue for now. The qemu library needs to be signed with the same certificate as the Amiberry bundle, otherwise MacOS will not allow it to load.

No way to codesign the plugin at user level? If not possible, the other option can be pull out the plugins of the pkg?

Edit: I tried to put the plugins outside the pkg, same codesign problem.

Edit2: this can't help? `codesign --force --deep -s - /Applcations/Emulatori/Amiberry.app/Contents/Resouces/plugins/qemu-uae.dylib`

**Comment by marmotta77** [comment]
New versions of Amiberry, same qemu issue.

**Reply by giantclambake** [reply]
Keep in mind, I've had very limited success with the plugin ~ see #1473  ; perhaps the qemu plugin is too old these days?
